.github/actions/setup-action/action.yml

```diff
@@ -0,0 +1,69 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+name: 'Setup action'
+description: 'Setup repository and configure the required steps'
+inputs:
+  comment_phrase:
+    description: "The comment phrase in the PR"
+    required: true
+  github_token:
+    description: "The GitHub token"
+    required: true
+  pull_request_url:
+    description: "The URL of the PR"
+    required: false
+    default: ''
+  github_repository:
+    description: "The GitHub repository"
+    required: false
+    default: ''
+  github_job:
+    description: "The GitHub job"
+    required: false
+    default: ''
+  github_current_run_id:
+    description: "The GitHub current run id. Not the same that is fetched in this action"
+    required: false
+    default: ''
+
+runs:
+  using: composite
+  steps:
+    - name: Check out repository code
+      uses: actions/checkout@v3
```

Comment:
Does having 2 checkout actions actually work? Does it check out 2 copies of the repo or does it actually overwrite the existing one?

I was anticipating this being a bash step like

`git fetch origin ${{ github.event.pull_request.head.sha }} && git checkout ${{ github.event.pull_request.head.sha }}`

Reply:
If it works, that's great, I'm just surprised

Reply:
Calling `actions/checkout` multiple times it updates/overwrites the same working directory with the latest changes from the specified branch or commit hash each time. It doesn't create multiple copies of the repository (until you explicitly specify it), and instead, it keeps the working directory up-to-date with the latest changes.

Reply:
Ok awesome!